quesma/queryparser/pancake_json_rendering.go

```diff
@@ -0,0 +1,180 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import (
+	"context"
+	"fmt"
+	"quesma/model"
+	"quesma/util"
+	"strings"
+)
+
+type pancakeJSONRenderer struct {
+}
+
+func (p *pancakeJSONRenderer) selectMetricRows(name string, rows []model.QueryResultRow) []model.QueryResultRow {
+	result := []model.QueryResultRow{}
+	for i, row := range rows {
+		newRow := model.QueryResultRow{Index: row.Index}
+		for _, col := range row.Cols {
+			if strings.HasPrefix(col.ColName, name) {
+				newRow.Cols = append(newRow.Cols, col)
+			}
+		}
+		result = append(result, newRow)
+		if i == 0 {
+			break // just one row please. FIXME Style. If here is only to fix staticcheck...
+		}
+	}
+	return result
+}
+
+func (p *pancakeJSONRenderer) splitBucketRows(name string, rows []model.QueryResultRow) ([]model.QueryResultRow, [][]model.QueryResultRow) {
+
+	buckets := []model.QueryResultRow{}
+	subAggrs := [][]model.QueryResultRow{}
+	if len(rows) == 0 {
+		return buckets, subAggrs
+	}
+	indexName := rows[0].Index
+	buckets = append(buckets, model.QueryResultRow{Index: indexName})
+	subAggrs = append(subAggrs, []model.QueryResultRow{{Index: indexName}})
+	for _, cols := range rows[0].Cols {
+
+		if strings.HasPrefix(cols.ColName, name+"key") || strings.HasPrefix(cols.ColName, name+"count") {
```

Comment:
I think we could extract consts for these strings like `key`, `count` as they are used in many places to avoid typos